File: univalued-binary-tree/univalued-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isUnivalTree(self, root: TreeNode) -> bool:
        '''
        Determines if the given binary tree only contains a single value
        Returns true if it does, otherwise false
        '''
        c1 = True
        c2 = True
        if root is not None:
            if root.left:
                if root.left.val == root.val:
                    c1 = self.isUnivalTree(root.left)
                else:
                    c1 = False
            if root.right:
                if root.right.val == root.val:
                    c2 = self.isUnivalTree(root.right)
                else:
                    c2 = False
        return c1 and c2

File: univalued-binary-tree/univalued-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution 
{
    public boolean isUnivalTree(TreeNode root) 
    {
        if (root == null) // No tree exists
            return true;
        
        int uniValue = root.val; //Every value should be root val
        Queue<TreeNode> queue = new LinkedList<TreeNode>();
        queue.add(root);
        while (queue.size() > 0)
        {
            TreeNode node = queue.remove();
            
            if (node.left != null)
                queue.add(node.left);
            if (node.right != null)
                queue.add(node.right);
            
            if (node.val != uniValue) // Not all the same
                return false;
        }
        return true;
    }
}